### PR TITLE
Made app init consistent and phased

### PR DESCRIPTION
**Pull request by damithc**
Made the init stages more natural, incremental.
Also eliminated the settings leak between manual testing and automated tests.

### CODE REVIEW

**Comment by damithc** [comment]
@m133225 @YijinL @HansNewbie have a look (this involves config, prefs, and model manager)

File: config.json [hunk 1]
```diff
@@ -1,9 +0,0 @@
-{
-  "appTitle" : "Address App",
-  "updateInterval" : 10000,
-  "simulateUnreliableNetwork" : true,
-  "currentLogLevel" : "DEBUG",
-  "specialLogLevels" : {
-    "SyncManager" : "TRACE"
-  }
-}
```
Comment (damithc) [inline config.json]: No need for this file to be tracked as the default config file is created automatically. Not tracking it allows devs to have their own config.

**Comment by HansNewbie** [comment]
Looks awesome 👍

File: src/main/java/address/model/ModelManager.java [hunk 2]
```diff
@@ -46,7 +46,8 @@
      * AddressBook and its variables should not be null.
      * @param src
      */
-    public ModelManager(AddressBook src, UserPrefs prefs) {
+
```
Comment (YijinL) [inline src/main/java/address/model/ModelManager.java]: Should there be a new line between the method declaration and it's header comment?

**Comment by YijinL** [comment]
Looks good

File: src/main/java/address/model/ModelManager.java (hunk 2 above)
Comment (damithc) [inline src/main/java/address/model/ModelManager.java]: No. got added during conflict resolution. I'll remove it.

**Pull request merged**